gears/src/baseapp/baseapp.rs

```diff
@@ -119,7 +119,12 @@ impl<
 
         //TODO: handle request height > 1 as is done in SDK
 
-        let mut ctx = InitContext::new(&mut multi_store, self.get_block_height(), request.chain_id);
+        let chain_id = request.chain_id.try_into().unwrap_or_else(|_| {
+            error!("Invalid chain id provided by Tendermint.\nTerminating process\n");
```

Comment:
What if we replace all `std::process::exit` with `expect` and use this crate for logs? 
https://docs.rs/log-panics/latest/log_panics/

Reply:
We need the node to stop. At the moment if we panic it's only the thread that dies. Given that we use tendermint-abci I'm not sure how we would get round that?